src/shared/components/case-editor/case-event-completion/components/case-event-completion-task-reassigned/case-event-completion-task-reassigned.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription, throwError } from 'rxjs';
+import { Task } from '../../../../../domain/work-allocation/Task';
+import { AlertService, SessionStorageService } from '../../../../../services';
+import { JudicialworkerService, WorkAllocationService } from '../../../services';
+import { CaseworkerService } from '../../../services/case-worker.service';
+import { CaseEventCompletionComponent, COMPONENT_PORTAL_INJECTION_TOKEN } from '../../case-event-completion.component';
+
+@Component({
+  selector: 'app-case-event-completion-task-reassigned',
+  templateUrl: './case-event-completion-task-reassigned.html'
+})
+export class CaseEventCompletionTaskReassignedComponent implements OnInit, OnDestroy {
+
+  public caseId: string;
+  public assignedUserId: string;
+  public assignedUserName: string;
+  public subscription: Subscription;
+  public caseworkerSubscription: Subscription;
+  public judicialworkerSubscription: Subscription;
+
+  constructor(@Inject(COMPONENT_PORTAL_INJECTION_TOKEN) private parentComponent: CaseEventCompletionComponent,
+    private readonly route: ActivatedRoute,
+    private readonly workAllocationService: WorkAllocationService,
+    private readonly sessionStorageService: SessionStorageService,
+    private readonly judicialworkerService: JudicialworkerService,
+    private readonly caseworkerService: CaseworkerService,
+    private readonly alertService: AlertService) {
+  }
+
+  public ngOnInit(): void {
+    // Get case id and task from the parent component
+    this.caseId = this.parentComponent.context.caseId;
+    const task = this.parentComponent.context.task;
+
+    // Current user is a caseworker?
+    this.caseworkerSubscription = this.caseworkerService.getCaseworkers(task.jurisdiction).subscribe(result => {
+      if (result && result[0].service === task.jurisdiction && result[0].caseworkers) {
+        const caseworker = result[0].caseworkers.find(x => x.idamId === task.assignee);
+        if (caseworker) {
+          this.assignedUserName = `${caseworker.firstName} ${caseworker.lastName}`;
+        }
+      }
+
+      if (!this.assignedUserName) {
+        // Current user is a judicial user?
+        this.judicialworkerSubscription = this.judicialworkerService.getJudicialworkers([task.assignee], task.jurisdiction)
+        .subscribe(judicialworkers => {
+          if (judicialworkers) {
+            const judicialworker = judicialworkers.find(x => x.sidam_id === task.assignee);
+            if (judicialworker) {
+              this.assignedUserName = judicialworker.full_name;
+            }
+          }
+
+          if (!this.assignedUserName) {
+            this.assignedUserName = 'another user';
+          }
+        });
+      }
+    });
+  }
+
+  public ngOnDestroy(): void {
+    if (this.subscription) {
```

Comment:
Minor comment - Think Phil went over some of these conditions and added a check of the closed flag to see if the subscription had been unsubscribed previously - possibly to help with tests. Might be worth doing that here? e.g. && !this.subscription.closed